lib/azure/armrest/model/base_model.rb

```diff
@@ -189,7 +190,9 @@ class ResourceProvider < BaseModel; end
     class Sku < BaseModel; end
     class Usage < BaseModel; end
 
-    class ResponseHeaders < BaseModel; end
+    class ResponseHeaders < BaseModel
+      undef_method :response_headers
```

Comment:
Not really, since I need the automatic method generation.

I think down the road we should actually create an ExtendedModel subclass that adds additional accessors like :response_headers, plus any other tweaks we need. Most models would then subclass ExtendedModel, while something like ResponseHeaders would just subclass BaseModel. But, I would ask that be a separate PR.